x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/InvalidateApiKeyRequest.java

```diff
@@ -65,17 +65,21 @@ public InvalidateApiKeyRequest(@Nullable String realmName, @Nullable String user
             throw new IllegalArgumentException("Must use either [id] or [ids], not both at the same time");
         }
 
-        this.realmName = realmName;
-        this.userName = userName;
-        if (id != null) {
-            this.ids = new String[] {id};
+        this.realmName = textOrNull(realmName);
+        this.userName = textOrNull(userName);
+        if (Strings.hasText(id)) {
+            this.ids = new String[]{id};
         } else {
             this.ids = ids;
         }
```

Comment:
Before I touched this class in 7.10, there is no check for `id` in either constructor. Last time I added
```
ids = Strings.hasText(id) == false ? null : new String[] { id };
```
in the constructor with `StreamInput`. Looking back now, this introduced a tiny inconsistency of how empty `id` is handled in different constructors. Your change here fixed this inconsistency. Thanks!

Now I have a new question: since a single empty string is now handled and excluded from the constructor, should we also apply the empty check for the string array of `ids`, i.e., should we move the [validation logic](https://github.com/elastic/elasticsearch/blob/7.10/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/action/InvalidateApiKeyRequest.java#L165-L175) of `ids` into here as well? On one hand, relocating the check for `ids` is good for consistency in that `validate()` is more about interactions between multiple fields. On the other hand, it complicates the constructors and also could be considered as a slight behaviour change. Overall, I think I prefer to not touch it.

Reply:
I took the cautious approach and decided to make this as close as possible to having zero visible changes.

Since the deserialisation code has a `hasText` check (as you note) I decided that this fix made sense, but chose not to try and filter empty values in the `ids` array. That seemed like a behavioural change that was outside the scope of this PR.

I'm not necessarily _opposed_ to someone making that change I just wanted this PR to be quick and simple.